keyfreq.sh

```diff
@@ -1,32 +1,42 @@
 #!/bin/bash
 
 
-# logs the key press frequency over 9 second window. Logs are written 
+# logs the key press frequency over 9 second window. Logs are written
 # in logs/keyfreqX.txt every 9 seconds, where X is unix timestamp of 7am of the
 # recording day.
 
 LANG=en_US.utf8
 
-helperfile="logs/keyfreqraw.txt" # temporary helper file
+helperfile='/dev/shm/keyfreqraw.txt'
+## FIXME: Still potential security risk even when not logged to persistent storage.
+## Maybe rewrite script in Python and count STDOUT directly and send SIGINT
+## after 9 seconds.
 
 mkdir -p logs
 
+trap 'kill $(jobs -p)' EXIT
+
 while true
 do
-  showkey > $helperfile &
-  PID=$!
-  
-  # work in windows of 9 seconds 
+  xinput test 11 > $helperfile &
```

Comment:
AFAIK when you have 2 keyboards (==laptop w. external one), you capture at most one with this.

For security purposes, I would rather pipe it to `| tr -d '[0-9]'` or so instead of `/dev/shm`